packages/calypso-router/src/index.js

```diff
@@ -1050,6 +1050,9 @@ function Context( path, state, pageInstance ) {
 	const window = this.page._window;
 	const hashbang = this.page._hashbang;
 
+	// merge multiple leading slashes into one, to avoid misinterpreting the path as scheme-relative URL
+	path = path.replace( /^\/+/, '/' );
```

Comment:
I think we might also need to handle backslashes, for example the code below will also redirect to `jetpack.com`
```
window.location.href = '/\\jetpack.com';
```

Reply:
Fixed in a9ed942 :+1: